### PR TITLE
 prevent switching order after changing between percent / number

### PR DESCRIPTION
**Pull request by dsnyder-fairway**

### CODE REVIEW

File: webapp/src/main/webapp/src/app/assessments/results/average-scale-score.component.ts [hunk 1]
```diff
@@ -24,6 +24,8 @@ export class AverageScaleScoreComponent {
 
   @Input()
   set statistics(value: ExamStatistics) {
+    value.percents = value.percents.reverse();
+    value.levels = value.levels.reverse();
```
Comment (malaffoon) [inline webapp/src/main/webapp/src/app/assessments/results/average-scale-score.component.ts]: Are the colors reversed as well? I wasn't sure from the last review whether that happened or not.

Is the order of these values assumed to be "ascending" anywhere else?

**Comment by dsnyder-fairway** [comment]
saw a comment to jira. Will have changes shortly

File: webapp/src/main/webapp/src/app/assessments/results/average-scale-score.component.ts (hunk 1 above)
Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/assessments/results/average-scale-score.component.ts]: colors were reversed in last review and this one. I just saw your comment to the jira. I'll give that a shot

Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/assessments/results/average-scale-score.component.ts]: the order of this is not used anywhere else

**Comment by malaffoon** [comment]
The pipe would only be used in the template, where the reverse() being called over and over was not appropriate. The way you have it (adjusting the order when setting the values) should work fine.

Again, assuming the order wasn't being relied upon elsewhere; which you confirmed.

File: webapp/src/main/webapp/src/app/assessments/results/average-scale-score.component.ts (hunk 1 above)
Comment (malaffoon) [reply webapp/src/main/webapp/src/app/assessments/results/average-scale-score.component.ts]: Might want to add a code comment that indicates why we're reversing them. Something like `// reverse so "good" comes before "bad"; that order is assumed in the UI`

File: webapp/src/main/webapp/src/app/assessments/results/average-scale-score.component.ts [hunk 2]
```diff
@@ -24,6 +24,9 @@ export class AverageScaleScoreComponent {
 
   @Input()
   set statistics(value: ExamStatistics) {
+    // reverse percents and levels so scale score statistics appear in descending order ("good" statistics levels comes before "bad")
+    value.percents = value.percents.reverse();
+    value.levels = value.levels.reverse();
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/assessments/results/average-scale-score.component.ts]: `Array.reverse()` is an in-place op. can you add a `.concat()` in between to make a copy of the array before reversing?

**Pull request merged**